src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -1386,25 +1386,6 @@ def end_string(self):
         # pylint: disable=no-self-use
         return "omp end target"
 
-    def validate_global_constraints(self):
```

Comment:
It's because a tree with a TargetDirective with a CodeBlock inside is not necessarily invalid. (but its the user that needs to guarantee if its valid)

For PSyclone the insert transformation validate still prevents putting them because it is probably the wrong thing to do. But if somebody goes out of its way to manually construct such tree, it should be allowed to the backend. There is a small number of these cases in NEMO that are needed for better performance.